src/models/euDccCertificates/index.ts

```diff
@@ -0,0 +1,122 @@
+import EuDccGenerator, {
+  BasicDetails,
+  TestingRecord,
+} from "@notarise-gov-sg/eu-dcc-generator";
+import { notarise, pdtHealthCertV2 } from "@govtechsg/oa-schemata";
+import _ from "lodash";
+import { Type } from "../fhir/constraints";
+import { ParsedBundle } from "../../models/fhir/types";
+import { isoToDateOnlyString, isoToLocaleString } from "../../common/datetime";
+import { config, getDefaultIfUndefined } from "../../config";
+import { getLogger } from "../../common/logger";
+
+const { trace } = getLogger("src/models/euDccCertificates");
+const { euSigner, testTypes } = config;
+
+const genEuDccCertificates = async (
+  type: Type,
+  parsedFhirBundle: ParsedBundle,
+  uuid: string,
+  storedUrl: string
+): Promise<notarise.SignedEuHealthCert[]> => {
+  const traceWithRef = trace.extend(`reference:${uuid}`);
+  const { PdtTypes } = pdtHealthCertV2;
+  let signedEuHealthCerts: notarise.SignedEuHealthCert[] = [];
+
+  if (
+    (_.isString(type) && (type === PdtTypes.Art || type === PdtTypes.Pcr)) ||
+    type.includes(PdtTypes.Art) ||
+    type.includes(PdtTypes.Pcr)
+  ) {
+    traceWithRef("signedEuHealthCerts: Generating EU test cert...");
+
+    const publicKey = getDefaultIfUndefined(
+      process.env.SIGNING_EU_QR_PUBLIC_KEY,
+      ""
+    );
+    const privateKey = getDefaultIfUndefined(
+      process.env.SIGNING_EU_QR_PRIVATE_KEY,
+      ""
+    );
+    const euDccGenerator = EuDccGenerator(
+      publicKey,
+      privateKey,
+      euSigner.issuer
+    );
+    const basicDetails: BasicDetails = {
+      reference: uuid,
+      issuerName: euSigner.issuer,
+      expiryDays: euSigner.expiryDays,
+      patientDetails: {
+        name: parsedFhirBundle.patient.fullName,
+        dateOfBirth: isoToDateOnlyString(parsedFhirBundle.patient.birthDate),
+        meta: {
+          reference: uuid,
+          notarisedOn: new Date().toISOString(),
+          passportNumber: parsedFhirBundle.patient.passportNumber,
+          url: storedUrl,
+        },
+      },
+    };
+
+    // Map the test type code from single healthcert oa-doc type
+    let testTypeCode: string;
+    if (type === PdtTypes.Art) {
+      testTypeCode = testTypes.ART;
+    } else if (type === PdtTypes.Pcr) {
+      testTypeCode = testTypes.PCR;
+    }
```

Comment:
We should put all mapping logic here.

It's not good that `testTypeCode` may still be undefined after this segment.

Reply:
it's handled in here https://github.com/Notarise-gov-sg/api-notarise-healthcerts/pull/487/files#diff-146b7700148dd936e26c0cbb8869b27b956c6addb56ed26541e1642170cedd47R74  intentionally leave this as undefined in here. if it undefined, will take observation data for multiple type cert.